### ISSUE TITLE
INVSTMTRS/REINVEST parse errors

### ISSUE THREAD

**Issue by PhracturedBlue** [libofx/libofx#13]
I'm trying to build an error-free investment OFX file, and I have several issues I cannot seem to remove.
I ran using the latest git code

To reproduce:
ofxdump test.ofx 2>&1 > /dev/null | grep -A1 ERROR
Result:
```
LibOFX ERROR: WRITEME: DTASOF (20180224) is not supported by the STATEMENT container
(Above message occurred on Line 23, Column 27)
--
LibOFX ERROR: WRITEME: INCOMETYPE (DIV) is not supported by the INVESTMENT container
(Above message occurred on Line 41, Column 34)
LibOFX ERROR: WRITEME: SUBACCTSEC (CASH) is not supported by the INVESTMENT container
(Above message occurred on Line 45, Column 35)
```

I believe all of these have valid syntax as per http://www.ofx.net/downloads/OFX%202.2.pdf

OFX File:
```xml
<?xml version="1.0" ?>
<?OFX OFXHEADER="200" VERSION="200" SECURITY="NONE" OLDFILEUID="NONE" NEWFILEUID="NONE"?>
<OFX>
  <SIGNONMSGSRSV1>
    <SONRS>
      <STATUS>
        <CODE>0</CODE>
        <SEVERITY>INFO</SEVERITY>
      </STATUS>
      <DTSERVER>20180223172905</DTSERVER>
      <LANGUAGE>ENG</LANGUAGE>
    </SONRS>
  </SIGNONMSGSRSV1>
  <INVSTMTMSGSRSV1>
      <INVSTMTTRNRS>
        <TRNUID>0</TRNUID>
        <STATUS>
          <CODE>0</CODE>
          <SEVERITY>INFO</SEVERITY>
        </STATUS>
        <INVSTMTRS>
          <DTASOF>20180224</DTASOF>
          <CURDEF>USD</CURDEF>
          <INVACCTFROM>
            <BROKERID>bankname</BROKERID>
            <ACCTID>1234-5678</ACCTID>
          </INVACCTFROM>
          <INVTRANLIST>
            <DTSTART>20180102</DTSTART>
            <DTEND>20180102</DTEND>
                <REINVEST>
                  <INVTRAN>
                    <FITID>06e2c3a14d60bfbfb43adec3efadf217</FITID>
                    <DTTRADE>20180102</DTTRADE>
                  </INVTRAN>
                  <SECID>
                    <UNIQUEID>ASDF</UNIQUEID>
                    <UNIQUEIDTYPE>TICKER</UNIQUEIDTYPE>
                  </SECID>
                  <INCOMETYPE>DIV</INCOMETYPE>
                  <TOTAL>
                    100.00
                  </TOTAL>
                  <SUBACCTSEC>CASH</SUBACCTSEC>
                  <UNITS>10.00</UNITS>
                  <UNITPRICE>10.00</UNITPRICE>
                </REINVEST>
          </INVTRANLIST>
        </INVSTMTRS>
      </INVSTMTTRNRS>
    
  </INVSTMTMSGSRSV1>
  <SECLISTMSGSRSV1>
    <SECLIST>
        <STOCKINFO>
          <SECINFO>
            <SECID>
              <UNIQUEID>ASDF</UNIQUEID>
              <UNIQUEIDTYPE>TICKER</UNIQUEIDTYPE>
            </SECID>
            <SECNAME>This is a test stock</SECNAME>
            <TICKER>ASDF</TICKER>
          </SECINFO>
        </STOCKINFO>
      
    </SECLIST>
  </SECLISTMSGSRSV1>
</OFX>
```

**Comment by benoitg** [comment]
That is correct, the elements are valid.  The WRITEME in the error message simply mean that those elements are not (yet) implemented in LibOFX.  Patches welcome...

**Comment by jeffjl74** [comment]
Using the ofxdump from my pending PR on the above example file yields:

PS D:\kmmapp> ofxdump issue_test.ofx
ofx_proc_status():
    Ofx entity this status is relevant to: SONRS
    Severity: INFO
    Code: 0, name: Success
    Description: The server successfully processed the request.

ofx_proc_status():
    Ofx entity this status is relevant to: INVSTMTTRNRS
    Severity: INFO
    Code: 0, name: Success
    Description: The server successfully processed the request.

ofx_proc_security():
    Unique ID of the security: ASDF
    Format of the Unique ID: TICKER
    Security type: STOCKINFO: Stock security
    Name of the security: This is a test stock
    Ticker symbol: ASDF

ofx_proc_account():
    Account ID: bankname 1234-5678
    Account name: Investment account 1234-5678 at broker bankname
    Account type: INVESTMENT
    Default Currency: USD
    Account #: 1234-5678

ofx_proc_statement():
    Default Currency: USD
    Account ID: bankname 1234-5678
    Statement as-of date: 02/24/18 11:59:00 Central Standard Time
    Start date of this statement: 01/02/18 11:59:00 Central Standard Time
    End date of this statement: 01/02/18 11:59:00 Central Standard Time

ofx_proc_transaction():
    Account ID : bankname 1234-5678
    Transaction type: OTHER: Other
    Date initiated: 01/02/18 11:59:00 Central Standard Time
    Total money amount: 100.00
    # of units: 10.0000 (shares)
    Unit price: 10.0000 (price per share)
    Financial institution's ID for this transaction: 06e2c3a14d60bfbfb43adec3efadf217
    Investment transaction type: REINVEST (Reinvestment of income)
    Unique ID of the security being traded: ASDF
    Format of the Unique ID: TICKER
    Income type: DIVIDEND
    Security sub account type: CASH
ofx_proc_security():
    Unique ID of the security: ASDF
    Format of the Unique ID: TICKER
    Security type: STOCKINFO: Stock security
    Name of the security: This is a test stock
    Ticker symbol: ASDF

**Comment by benoitg** [comment]
This has been implemented in #19
